lib/auth/auth_with_roles.go

```diff
@@ -6409,11 +6409,32 @@ func (a *ServerWithRoles) GetWindowsDesktops(ctx context.Context, filter types.W
 	return filtered, nil
 }
 
+// hostIDExistsAndIsValid checks if the host ID exists and matches a registered windows desktop service.
+func (a *ServerWithRoles) hostIDExistsAndIsValid(ctx context.Context, s types.WindowsDesktop) error {
```

Comment:
Host ID is not a desktop-specific concept, and this method is placed on a rather generic `ServerWithRoles`. I would rename this.

Reply:
I made this method name specific to windows desktops: 106c26f4b916c4108b57be8b9f5f84974f8c595d